master/internal/command/command.go

```diff
@@ -146,357 +139,326 @@ func remakeCommandsByType(
+	if err := c.db.CompleteTask(c.taskID, time.Now().UTC()); err != nil {
+		c.syslog.WithError(err).Error("marking task complete")
+	}
 
-	case *apiv1.SetShellPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
+	if err := user.DeleteSessionByToken(ctx, c.GenericCommandSpec.Base.UserSessionToken); err != nil {
+		c.syslog.WithError(err).Errorf("failure to delete user session for task: %v", c.taskID)
+	}
+
+	go func() {
+		time.Sleep(terminatedDuration)
+		c.garbageCollect(ctx)
+	}()
+}
+
+// gc garbage collects the exited command.
+func (c *command) garbageCollect(ctx context.Context) {
+	if err := tasklist.GroupPriorityChangeRegistry.Delete(c.jobID); err != nil {
+		c.syslog.WithError(err).Error("deleting command from GroupPriorityChangeRegistry")
+	}
+	if c.exitStatus == nil {
+		if err := c.db.CompleteTask(c.taskID, time.Now().UTC()); err != nil {
+			c.syslog.WithError(err).Error("marking task complete")
 		}
-		ctx.Respond(&apiv1.SetShellPriorityResponse{Shell: c.toShell(ctx)})
+	}
 
-	case *tensorboardv1.Tensorboard:
-		ctx.Respond(c.toTensorboard(ctx))
+	if err := user.DeleteSessionByToken(ctx, c.GenericCommandSpec.Base.UserSessionToken); err != nil {
+		c.syslog.WithError(err).Errorf(
+			"failure to delete user session for task: %v", c.taskID)
+	}
 
-	case *apiv1.GetTensorboardRequest:
-		ctx.Respond(&apiv1.GetTensorboardResponse{
-			Tensorboard: c.toTensorboard(ctx),
-			Config:      protoutils.ToStruct(c.Config),
-		})
+	jobservice.DefaultService.UnregisterJob(c.jobID)
+	DefaultCmdService.unregisterCommand(c.taskID)
+}
 
-	case *apiv1.KillTensorboardRequest:
-		err := task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
-		if err != nil {
-			ctx.Log().WithError(err).Warn("failed to kill allocation")
-		}
-		ctx.Respond(&apiv1.KillTensorboardResponse{Tensorboard: c.toTensorboard(ctx)})
+func (cs *commandService) unregisterCommand(id model.TaskID) {
+	cs.mu.Lock()
+	defer cs.mu.Lock()
 
-	case *apiv1.SetTensorboardPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
-		}
-		ctx.Respond(&apiv1.SetTensorboardPriorityResponse{Tensorboard: c.toTensorboard(ctx)})
-
-	case *apiv1.DeleteWorkspaceRequest:
-		if c.Metadata.WorkspaceID == model.AccessScopeID(msg.Id) {
-			err := task.DefaultService.Signal(
-				c.allocationID,
-				task.KillAllocation,
-				"user requested workspace delete",
-			)
-			if err != nil {
-				ctx.Log().WithError(err).Warn("failed to kill allocation while deleting workspace")
-			}
-		}
+	delete(cs.commands, id)
+}
 
-	case sproto.ContainerLog:
+// commandService NTSC methods: LaunchNTSC, GetNTSC, listNTSCType, DeleteWorkspaceNTSC
 
-	case terminateForGC:
-		ctx.Self().Stop()
+// launchNTSC launches a generic NTSC command & returns the command.
+// launchNTSC -> CreateGenericCommand -> startCmd() ...
+func (cs *commandService) launchNTSC(
```

Comment:
i don't see this method as very helpful, as opposed to users just calling `createGenericCommand` directly